keps/prod-readiness/sig-network/4004.yaml

```diff
@@ -0,0 +1,6 @@
+# The KEP must have an approver from the
+# "prod-readiness-approvers" group 
+# of http://git.k8s.io/enhancements/OWNERS_ALIASES
+kep-number: 4004
+alpha:
+  approver: "@wojtek-t"
```

Comment:
Few comments:

0) High-level question: for an existing node with the field set, after enabling feature-gate will kubelet set the field to empty?
Can you add an answer to this question explicitly to the KEP (in proposal section I guess)?

1) Testing strategy:
- the test I would additionally like to see (I guess it may be manual and it could probably be the one that would be effective the test in https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/4004-deprecate-kube-proxy-version#were-upgrade-and-rollback-tested-was-the-upgrade-downgrade-upgrade-path-tested ) is a test that ensures that after I disable the FG the field will be again propagated.

2) Feature gates:
- just double checking - won't kube-apiserver also depend on the FG? [I think it shouldn't and if someone sets the field it should accept it, but wanted to double check]

3) feature enablement/disablement tests

I only partially agree with the answer - I have two concerns:
- depending on the answer to my (0) question - we may want to actually verify if kubelet zero's the field
- independently of everything, I would like to see some test that ensures that after disablement of the FG (given the reason of that will most probably be "oops, something still depends on it"), ensure that this will actually be set

[I will have some additional comments for Beta, but I want to unblock Alpha first :) ]

Reply:
>     * just double checking - won't kube-apiserver also depend on the FG? [I think it shouldn't and if someone sets the field it should accept it, but wanted to double check]

We weren't planning to actually block access to the field. Just to have kubelet stop setting it. So no apiserver involvement.

Reply:
> 1. for an existing node with the field set, after enabling feature-gate will kubelet set the field to empty

I think with feature-gate enabled, the kubelet sets the fields to empty.